tests/common/plugins/custom_skipif/__init__.py

```diff
@@ -0,0 +1,217 @@
+            show_platform_summary_raw_output = subprocess.check_output(show_platform_cmd, shell=True)
+            for line in show_platform_summary_raw_output.splitlines():
+                if 'Platform:' in line:
+                    platform_type = line.split()[1:][0]  # get platform, example: x86_64-mlnx_msn2700-r0
+                    item.session.config.cache.set(dynamic_tests_skip_platform_type, platform_type)
+                    break
+        except Exception as err:
+            logger.error('Unable to get platform type. Custom skip by platform impossible. Error: {}'.format(err))
+
+    logger.debug('Current platform type is: {}'.format(platform_type))
+    return platform_type
+
+
+def get_inventory_argument(inventory):
+    """Get Ansible inventory arguments"""
+    inv = ''
+    for inv_item in inventory.split(','):
+        inv += ' -i {}'.format(inv_item)
+    return inv
+
+
+def read_skip_file(item, skip_tests_file_path):
+    """
+    Read yaml file with list of test cases which should be skipped
+    :param item: pytest test item
+    :param skip_tests_file_path: path to file where stored list of test cases which should be skipped
+    :return: yaml loaded dictionary
+    """
+    custom_skip_if_dict = 'custom_skip_if_dict'
+    skip_dictionary = item.session.config.cache.get(custom_skip_if_dict, None)
+    if not skip_dictionary:
+        with open(skip_tests_file_path) as skip_data:
+            logger.debug('Reading dynamic skip file: {}'.format(skip_tests_file_path))
+            skip_dictionary = yaml.load(skip_data, Loader=yaml.FullLoader)
+            item.session.config.cache.set(custom_skip_if_dict, skip_dictionary)
+    return skip_dictionary
+
+
+def get_tests_to_be_skipped_path(skip_tests_file='tests_to_be_skipped_conditionally.yaml'):
+    """
+    Get path to file with dynamic skip information
+    :param skip_tests_file: skip test file name
+    :return: full path to skip test file name
+    """
+    custom_skip_folder_path = os.path.dirname(__file__)
+    custom_skip_tests_file_path = os.path.join(custom_skip_folder_path, skip_tests_file)
+    return custom_skip_tests_file_path
+
+
+def update_syspath_for_dynamic_import():
+    """
+    Update sys.path by current folder to have possibility to load python modules dynamically
+    """
+    if os.path.dirname(__file__) not in sys.path:
+        sys.path.append(os.path.dirname(__file__))
+
+
+def prepare_checkers(skip_dict, current_platform):
+    """
+    Import dynamically checker modules and initialize them
+    :param skip_dict: dictionary with skip test case skip conditions
+    :param current_platform: string with current platform, example: x86_64-mlnx_msn2700-r0
+    :return: list with checkers objects
+    """
+    skip_checkers_list = []
+    extra_params = {'current_platform': current_platform}
+    for skip_by in skip_dict:
+        logger.debug('Importing dynamic skip module: {}'.format(skip_by))
+        try:
+            skip_module_obj = eval(
```

Comment:
Running eval in this manner is generally considered poor python practice. I think you should be capable of doing this without calling `eval()` and I would suggest you refactor this line to do so. 

I have found personally at least eval tends to attract bugs and introduce undefined behavior.

Reply:
eval() allow to do import dynamically.
If not use eval() we will not be able to import modules dynamically.
Do you have any suggestion how to implement this logic in another way?

Reply:
here's an example: https://stackoverflow.com/questions/301134/how-to-import-a-module-given-its-name-as-string